### ISSUE TITLE
MiddleWare throwing error!

### ISSUE THREAD

**Issue by rsreevishal** [cthwaite/fastapi-websocket-broadcast#1]
**ERROR**

File "./app.py", line 189, in <module>
    app.add_middleware(RoomEventMiddleware)
  File "/home/sree/SREEVISHAL/Project/fastapi-websocket-broadcast/venv/lib/python3.6/site-packages/starlette/applications.py", line 117, in add_middleware
    self.middleware_stack = self.build_middleware_stack()
  File "/home/sree/SREEVISHAL/Project/fastapi-websocket-broadcast/venv/lib/python3.6/site-packages/starlette/applications.py", line 81, in build_middleware_stack
    app = cls(app=app, **options)
TypeError: __init__() got an unexpected keyword argument 'app'

**My package versions**

- fastapi==0.58.0
- starlette==0.13.4
- uvicorn==0.11.5

**Command used to run the app**

uvicorn app:app

**Comment by rsreevishal** [comment]
I see the pip file

> [[source]]
> name = "pypi"
> url = "https://pypi.org/simple"
> verify_ssl = true
> 
> [packages]
> aiofiles = "==0.4.0"
> fastapi = "==0.42.0"
> uvicorn = "==0.10.3"
> 
> [dev-packages]
> black = "*"
> pylint = "*"
> 
> [requires]
> python_version = "3.7"
> 
> [pipenv]
> allow_prereleases = true

maybe is it the version issue??

**Comment by dreamfalcon** [comment]
Same problem here.
Probably a version issue, but no idea if is a bug or change in the framework.

**Comment by dreamfalcon** [comment]
Fixed:
```
class RoomEventMiddleware:  # pylint: disable=too-few-public-methods
    """Middleware for providing a global :class:`~.Room` instance to both HTTP
    and WebSocket scopes.

    Although it might seem odd to load the broadcast interface like this (as
    opposed to, e.g. providing a global) this both mimics the pattern
    established by starlette's existing DatabaseMiddlware, and describes a
    pattern for installing an arbitrary broadcast backend (Redis PUB-SUB,
    Postgres LISTEN/NOTIFY, etc) and providing it at the level of an individual
    request.
    """

    def __init__(self, app):
        self._app = app
        self._room = Room()

    async def __call__(self, scope: Scope, receive: Receive, send: Send):
        if scope["type"] in ("lifespan", "http", "websocket"):
            scope["room"] = self._room
        await self._app(scope, receive, send)


app.add_middleware(RoomEventMiddleware)
```

**Comment by kthwaite** [comment]
Hi, thanks for opening!

Looks like this is a version issue - because the demo was written to work with the pinned dependencies in the `Pipenv` file. The versions you list above are much newer than this. Inbetween these versions, Starlette changed the way that Middleware was initialized, and now passes `app` as a keyword argument (https://github.com/encode/starlette/blob/master/starlette/applications.py#L90).

I've applied the correction (thanks, @dreamfalcon!) and updated the pinned deps in `master`. Closing the issue.
